### DISCUSSION TITLE
How to do I encrypt the Url in  edit action.

### DISCUSSION THREAD

**Discussion by mani1430** [Power-Components/livewire-powergrid#397] [Q&A] [answered]
1. How I add the $loop->index values in ID column. I don't want to use database table primary id values in the table row number.

2. I need to use encrypted URL , So please help me to how to encrypt the edit route parameter.

please help me.

like => Button::add('edit')
->target('')
->caption('')
->class(' cursor-pointer text-sky-600 px-3 py-2.5 m-1 rounded text-sm')
->route('customer.edit', ['url' => [ 'id' , Crypt::encrypt($id);]]),

how I don't have any idea , please help me.

3. Need to place Actions buttons near to id columns in the table.

**Comment by luanfreitasdev** [comment, marked as answer]
```php
public function addColumns(): PowerGridEloquent
    {
        return PowerGrid::eloquent()
            ->addColumn('id')
            ->addColumn('encrypt', fn (Customer $customer) => Crypt::encrypt($customer->id))

...

Button::add('edit')
        ->caption('Edit')
        ->route('customer-edit', ['id' => 'encrypt']),
```

**Reply by mani1430** [reply]
Thanks for the answer
